core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ShaclSail.java

```diff
@@ -972,7 +971,8 @@ public IsolationLevel getDefaultIsolationLevel() {
 
 			Disabled("Disabled"),
 			Auto("Auto"),
-			Bulk("Bulk");
+			Bulk("Bulk"),
+			BulkParallelCache("BulkParallelCache");
```

Comment:
@jeenbroekstra What do you think about this name. `Bulk` is used for typical bulk operations like loading a lot of data. Since that much data could make us run out of memory, but parallel validation and caching is disabled. But if you do have the memory, then running the bulk validation in parallel with cache will be up to 2.5x faster (in our benchmark). Question is, what do you think of the naming?

Reply:
I have no immediate objection to the naming itself, but I am a little confused how this combines with `setParallelValidation`, and more generally: instead of creating a new enums for combinations of settings (Bulk + Parallel + Caching), shouldn't we perhaps make Parallel and Caching enum options themselves?

Reply:
Might make for sense to make separate enums yes. The `Bulk` enum currently overrides any sail level settings like `setParallelValidation(...)`.

Reply:
Decided to not solve this here. So I've removed BulkParallelCache.